drivers/modem/modem_iface_uart_async.c

```diff
@@ -61,6 +61,17 @@ static void iface_uart_async_callback(const struct device *dev,
 		/* Notify upper layer that new data has arrived */
 		k_sem_give(&data->rx_sem);
 		break;
+	case UART_RX_STOPPED:
+		break;
+	case UART_RX_DISABLED:
+		/* RX stopped (likely due to line error), re-enable it */
+		k_mem_slab_alloc(&uart_modem_async_rx_slab, (void **)&buf, K_FOREVER);
```

Comment:
Following the UART API documentation https://docs.zephyrproject.org/latest/hardware/peripherals/uart.html, UART_RX_DISABLED could mean that there was no buffer available for the UART to switch to.

Please validate the return code of `k_mem_slab_alloc()` to ensure a buffer could be allocated before passing it to the UART as is done in the UART_RX_BUF_REQUEST switch case.